apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { Button } from "../core";
+
+type ImportState =
+  | {
+      valid: true;
+      imported: false;
+      pcds: PCD[];
+      folders: FolderMap;
+      conflicts: number;
+    }
+  | { valid: true; imported: true; added: number }
+  | { valid: false };
+
+type FolderMap = { [pcdId: string]: string };
+
+export function AccountImportModal() {
```

Comment:
User communication question: Is this clearer as "Import" or "Restore from Backup"?

Reply:
There are actually multiple ways this could work which might suggest different words.  n theory we could give the user choices between them.

- Import if new: This is how this works today.
- Import and replace: Handle conflicting IDs by keeping the imported one.  This might be more of a "restore" operation.
- Replace full state: We could delete PCDs not included in the import.  This would be more of a "rollback" operation, to go back to a prior state.

Note this is all a discussion of a future version of this feature, not complexity to add today.  I think "import" actually does best describe the current functionality.